include/fiction/algorithms/simulation/sidb/displacement_robustness.hpp

```diff
@@ -0,0 +1,550 @@
+            for (std::size_t i = 0; i < cell_displacements.size(); ++i)
+            {
+                lyt.assign_cell_type(cell_displacements[i], layout.get_cell_type(sidbs_of_the_original_layout[i]));
+                if (!layout.is_empty_cell(cell_displacements[i]))
+                {
+                    identical_cells_to_original_layout++;
+                }
+            }
+            if (lyt.num_cells() == layout.num_cells() && identical_cells_to_original_layout != layout.num_cells())
+            {
+                layouts.push_back(lyt);
+            }
+            number_of_layouts_with_displaced_sidbs++;
+        }
+        if (layouts.empty())
+        {
+            throw std::runtime_error("No valid displaced SiDB layouts could be generated.");
+        }
+        return layouts;
+    }
+    /**
+     * The SiDB cell-level layout/SiDB gate for which the displacement robustness calculation is performed.
+     */
+    const Lyt& layout;
+    /**
+     * The parameters for the displacement robustness computation.
+     */
+    sidb_gate_displacement_robustness_params<Lyt>& params;
+    /**
+     * The statistics of the displacement robustness computation.
+     */
+    displacement_robustness_domain_stats& stats;
+    /**
+     * This stores all possible displacements for all SiDBs in the SiDB layout. This means e.g. the first vector
+     * describes all possible positions of the first SiDB due to the allowed/possible displacements.
+     */
+    std::vector<std::vector<cell<Lyt>>> all_possible_displacements_for_each_sidb{};
+    /**
+     * SiDB positions of the originally given SiDB layout.
+     */
+    std::vector<cell<Lyt>> sidbs_of_the_original_layout{};
+    /**
+     * The specification of the layout.
+     */
+    const std::vector<TT>& truth_table;
+};
+
+}  // namespace detail
+
+/**
+ * This function determines the operational status of all possible displacements of the SiDBs in the SiDB gate,
+ * based on the provided truth table specification and displacement robustness computation parameters.
+ * The number of displacements grows exponentially with the number of SiDBs. For small layouts, all displacements
+ * can be analyzed. For larger layouts, random sampling can be applied, controllable by the ``analysis_mode`` and
+ * ``percentage_of_analyzed_displaced_layouts`` in sidb_gate_displacement_robustness_params.
+ *
+ * @tparam Lyt The SiDB cell-level layout type.
+ * @tparam TT The type of the truth table specifying the gate behavior.
+ * @param truth_table_spec Vector of truth table specifications.
+ * @param params Parameters for the displacement robustness computation.
+ * @param stats Statistics related to the displacement robustness computation.
+ * @return The displacement robustness domain of the SiDB gate.
+ */
+template <typename Lyt, typename TT>
+[[nodiscard]] displacement_robustness_domain<Lyt>
+determine_sidb_gate_displace_robustness_domain(const Lyt& layout, const std::vector<TT>& spec,
```

Comment:
I agree that we should use ``displacement`` here. However, since we have two functions in this header-file, it is difficult to name the function as the filename, right?

Reply:
I agree. Sometimes there are helper functions that are called something different. However, this should not stop us from naming the main function in accordance with the file name.